src/common.h

```diff
@@ -27,6 +29,47 @@
 
 #define MASK(n) (~((~0U << (n))))
 
+#if defined(_MSC_VER)
+#include <intrin.h>
+static inline int clz(uint32_t v)
+{
+    uint32_t leading_zero = 0;
+    if (_BitScanReverse(&leading_zero, v))
+        return 31 - leading_zero;
+    return 32; /* undefined behavior */
+}
+#elif defined(__GNUC__) || defined(__clang__)
+static inline int clz(uint32_t v)
+{
+    return __builtin_clz(v);
```

Comment:
> Returns the number of leading 0-bits in x, starting at the most significant bit position. If x is 0, the result is undefined. 
Quote from [here](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html)

Should we perform a bitwise OR with 1 on the input, as the GNU `__builtin_clz` documentation states that the result is undefined if x is 0? Alternatively, should we include an assertion to prevent 0 as an input?

Reply:
> Should we perform a bitwise OR with 1 on the input, as the GNU `__builtin_clz` documentation states that the result is undefined if x is 0? Alternatively, should we include an assertion to prevent 0 as an input?

Since (rv_)clz is used internally, you can simply document it with caution.